tests/v2/actions/charts/stackstate.go

```diff
@@ -0,0 +1,411 @@
+	kubenamespaces "github.com/rancher/rancher/tests/v2/actions/kubeapi/namespaces"
+	"github.com/rancher/rancher/tests/v2/actions/namespaces"
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/extensions/defaults"
+	"github.com/rancher/shepherd/pkg/api/steve/catalog/types"
+	"github.com/rancher/shepherd/pkg/wait"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+const (
+	StackstateExtensionNamespace = "cattle-ui-plugin-system"
+	StackstateExtensionsName     = "observability"
+	UIPluginName                 = "rancher-ui-plugins"
+	StackstateK8sAgent           = "stackstate-k8s-agent"
+	StackstateNamespace          = "stackstate"
+	StackstateCRD                = "observability.rancher.io.configuration"
+	local                        = "local"
+	RancherPartnerChartRepo      = "rancher-partner-charts"
+)
+
+// InstallStackstateExtension is a helper function that installs the stackstate extension chart.
+func InstallStackstateExtension(client *rancher.Client, installExtensionOptions *ExtensionOptions) error {
+
+	extensionInstallAction := newStackstateExtensionsInstallAction(installExtensionOptions)
+
+	catalogClient, err := client.GetClusterCatalogClient(local)
+	if err != nil {
+		return err
+	}
+
+	// register uninstall stackstate extension as a cleanup function
+	client.Session.RegisterCleanupFunc(func() error {
+		// UninstallAction for when uninstalling the stackstate agent chart
+		defaultChartUninstallAction := newChartUninstallAction()
+
+		err := catalogClient.UninstallChart(StackstateExtensionsName, StackstateExtensionNamespace, defaultChartUninstallAction)
+		if err != nil {
+			return err
+		}
+		log.Info("Uninstalled observability extension successfully.")
+
+		watchAppInterface, err := catalogClient.Apps(StackstateExtensionNamespace).Watch(context.TODO(), metav1.ListOptions{
+			FieldSelector:  "metadata.name=" + StackstateExtensionsName,
+			TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+		})
+		if err != nil {
+			return err
+		}
+
+		err = wait.WatchWait(watchAppInterface, func(event watch.Event) (ready bool, err error) {
+			chart := event.Object.(*catalogv1.App)
+			if event.Type == watch.Error {
+				return false, fmt.Errorf("there was an error uninstalling stackstate extension")
+			} else if event.Type == watch.Deleted {
+				return true, nil
+			} else if chart == nil {
+				return true, nil
+			}
+			return false, nil
+		})
+		if err != nil {
+			return err
+		}
+
+		err = catalogClient.UninstallChart(StackstateExtensionsName, StackstateExtensionNamespace, defaultChartUninstallAction)
+		if err != nil {
+			return err
+		}
+
+		steveclient, err := client.Steve.ProxyDownstream(local)
+		if err != nil {
+			return err
+		}
+		namespaceClient := steveclient.SteveType(namespaces.NamespaceSteveType)
+
+		namespace, err := namespaceClient.ByID(StackstateExtensionNamespace)
+		if err != nil {
+			return err
+		}
+
+		err = namespaceClient.Delete(namespace)
```

Comment:
This namespace is created when the extension is installed. Since it is created as part of the test, we ensure that the namespace is deleted as well during cleanup.